### PR TITLE
Nastya

### PR DESCRIPTION
**Pull request by AnastasiyaStr**
Optimised update method

### CODE REVIEW

**Review by AndriY2389 — APPROVED** [review]

File: src/main/java/com/softserve/demo/controller/CustomerController.java [hunk 1]
```diff
@@ -1,98 +1,83 @@
...
-//     @PostMapping("saveUserProfile")
-//     public ResponseEntity<?> createCustomer(@RequestParam("file")MultipartFile multipartFile,
-//        @RequestParam("user")String user
-//     )
-//         return new ResponseEntity<>(HttpStatus.CREATED);
-//     }
+    private final CustomerService customerService;
+    private final FilesStorageService fileStorageService;
 
+    @Autowired
```
Comment (SilichAlex) [inline src/main/java/com/softserve/demo/controller/CustomerController.java:27]: Лишній Autowired

File: src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java [hunk 2]
```diff
@@ -1,100 +1,93 @@
...
+import java.util.stream.Collectors;
 
 @Service
 public class CustomerServiceImpl implements CustomerService {
-@Autowired
-CustomerRepository customerRepository;
+    private final CustomerRepository customerRepository;
+    private final UserRepository userRepository;
+
     @Autowired
```
Comment (SilichAlex) [inline src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java:22]: Те саме

File: src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java [hunk 3]
```diff
@@ -1,100 +1,93 @@
...
+
     private static java.sql.Date convertUtilToSql(java.util.Date uDate) {
-        java.sql.Date sDate = new java.sql.Date(uDate.getTime());
-        return sDate;
+        return new java.sql.Date(uDate.getTime());
     }
 
     @Override
-    public Customer updateCustomer(Integer id, Customer customer) {
+    public CustomerDTO updateCustomer(Integer id, CustomerDTO customerDTO) {
```
Comment (SilichAlex) [inline src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java:43]: Чому не void

File: src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java [hunk 4]
```diff
@@ -1,100 +1,93 @@
...
-        List<Customer> customers = customerRepository.findAll();
-        return customers;
+    public List<CustomerDTO> getAllCustomers() {
+        return customerRepository.findAll().stream().map(
+                CustomerMapper.INSTANCE::CustomerToCustomerDTO).collect(Collectors.toList());
     }
 
     @Override
-    public Customer deleteCustomer(Integer id) {
+    public CustomerDTO deleteCustomer(Integer id) {
```
Comment (SilichAlex) [inline src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java:64]: Повертати void

File: src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java [hunk 5]
```diff
@@ -1,100 +1,93 @@
...
         if (!customerRepository.existsById(id)) {
             return null;
         }
-        Customer customer =  customerRepository.findById(id).get();
-       customerRepository.deleteById(id);
-        return customer;
+        Customer customer = customerRepository.findById(id).get();
+        customerRepository.deleteById(id);
+        return CustomerMapper.INSTANCE.CustomerToCustomerDTO(customer);
```
Comment (DimaNarepeha) [inline src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java:70]: you can extract CustomerMapper.INSTANCE  in a separate private final field. And spring will inject it

**Pull request closed without merging**